provider/pkg/provider/provider.go

```diff
@@ -23,30 +23,27 @@ import (
 
 func construct(ctx *pulumi.Context, typ, name string, inputs provider.ConstructInputs,
 	options pulumi.ResourceOption) (*provider.ConstructResult, error) {
-	// TODO: Add support for additional component resources here.
 	switch typ {
-	case "xyz:index:StaticPage":
-		return constructStaticPage(ctx, name, inputs, options)
+	case "catalystsquad-platform":
```

Comment:
I feel like the difference here is going to matter, but I don't have a technical reason for why at the moment, so, just calling it out for thought.

Reply:
oh this is actually a mistake. I found it when I started testing with the generated sdk that this PR created, and I ran into a:
```
error: program failed: waiting for RPCs: rpc error: code = Unknown desc = unknown resource type catalystsquad-platform:index:Vpc
```
oops.

Anyways, should be fixed now along with another thing. Your intuition was correct.